pr_agent/algo/utils.py

```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import json
+import logging
+import re
 import textwrap
 
```

Comment:
Consider adding a default value for the `max_iter` parameter in the function signature. This would make the function more flexible and allow the caller to decide on the maximum number of iterations if needed. [medium]